src/ray/raylet/raylet.cc

```diff
@@ -79,6 +79,9 @@ Raylet::Raylet(instrumented_io_context &main_service, const std::string &socket_
   self_node_info_.set_node_manager_port(node_manager_.GetServerPort());
   self_node_info_.set_node_manager_hostname(boost::asio::ip::host_name());
   self_node_info_.set_metrics_export_port(metrics_export_port);
+  const auto &resource_map = node_manager_config.resource_config.GetResourceMap();
+  self_node_info_.mutable_resources_total()->insert(resource_map.begin(),
```

Comment:
Previously, since we didn't report node resource information, when a new node is reported to GCS, and pg manager tries to reschedule infeasible pgs, it always failed. 

The resource is currently reported "again" by line 122, which is pretty weird. I believe it is the legacy code path we should clean. cc @iycheng (We should remove the Resource Table from GCS)

Reply:
I agree. The resource table seems like something can be reconstructed by just pulling data.